src/webgpu/api/operation/adapter/requestDevice_limits.spec.ts

```diff
@@ -0,0 +1,148 @@
+export const description = `
+Tests passing various requiredLimits to GPUAdapter.requestDevice.
+`;
+
+import { Fixture } from '../../../../common/framework/fixture.js';
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../common/util/data_tables.js';
+import { assert } from '../../../../common/util/util.js';
+import { DefaultLimits } from '../../../constants.js';
+
+const kLimitTypes = keysOf(DefaultLimits);
+
+const kMaxUnsignedLongValue = 4294967295;
+/** Clamps a numeric value to the valid unsigned long range, as defined by WebIDL */
+function clampToUnsignedLong(value: number): number {
+  return Math.min(kMaxUnsignedLongValue, Math.max(0, value));
+}
+
+export const g = makeTestGroup(Fixture);
+
+g.test('unknown_limits')
+  .desc(
+    `
+    Test that specifiying limits that aren't part of the supported limit set causes
+    requestDevice to reject.`
+  )
+  .fn(async t => {
+    const adapter = await navigator.gpu.requestAdapter();
+    assert(adapter !== null);
+
+    const requiredLimits: Record<string, number> = { unknownLimitName: 9000 };
+
+    t.shouldReject('OperationError', adapter.requestDevice({ requiredLimits }));
+  });
+
+g.test('default_limits')
+  .desc(
+    `
+    Test that each supported limit can be specified with the default values required by the
+    spec.
+    - Tests each limit`
+  )
+  .paramsSubcasesOnly(u =>
+    u.combine('limit', kLimitTypes).unless(p => typeof DefaultLimits[p.limit] !== 'number')
+  )
+  .fn(async t => {
+    const { limit } = t.params;
+
+    const adapter = await navigator.gpu.requestAdapter();
+    assert(adapter !== null);
+
+    const requiredLimits: Record<string, number> = {};
+    requiredLimits[limit] = DefaultLimits[limit];
```

Comment:
uNit: I believe something like this would work:
```suggestion
    const requiredLimits = { [limit]: DefaultLimits[limit] } as const;
```